### PR TITLE
fna3d: init at 24.11

### PR DESCRIPTION
**Pull request by mrtnvgr**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/fn/fna3d/package.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+, stdenv
+, fetchgit
+, cmake
+, SDL2
+}:
+stdenv.mkDerivation rec {
+  pname = "fna3d";
+  version = "23.11";
+
+  src = fetchgit {
```
Comment (MikaelFangel) [inline pkgs/by-name/fn/fna3d/package.nix:11]: Should be fetchFromGithub (:

File: pkgs/by-name/fn/fna3d/package.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    ln -s $out/lib/libFNA3D.so $out/lib/libFNA3D.so.0.${version}
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "Accuracy-focused XNA4 reimplementation for open platforms";
+    homepage = "https://fna-xna.github.io/";
+    license = licenses.mspl;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ mrtnvgr ];
```
Comment (MikaelFangel) [inline pkgs/by-name/fn/fna3d/package.nix:32]: mainProgram is missing from the meta. Please add it, as the old behaviour is deprecated.

File: pkgs/by-name/fn/fna3d/package.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchFromGitHub {
+    owner = "FNA-XNA";
+    repo = "FNA3D";
+    rev = version;
+    fetchSubmodules = true;
+    hash = "sha256-UXphZQrk+MUOcUIqZ0ho3HTP5vNZjN9ifsANOJbTfVE=";
+  };
+
+  buildInputs = [ cmake SDL2 ];
```
Comment (Scrumplex) [inline pkgs/by-name/fn/fna3d/package.nix:19]: Dependencies that need to be executed at build time have to go in `nativeBuildInputs`

```suggestion
  nativeBuildInputs = [ cmake ];
  
  buildInputs = [ SDL2 ];
```

File: pkgs/by-name/fn/fna3d/package.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+  installPhase = ''
+    runHook preInstall
+    install -Dm755 libFNA3D.so $out/lib/libFNA3D.so
+    ln -s $out/lib/libFNA3D.so $out/lib/libFNA3D.so.0
+    ln -s $out/lib/libFNA3D.so $out/lib/libFNA3D.so.0.${version}
```
Comment (Scrumplex) [inline pkgs/by-name/fn/fna3d/package.nix:25]: I think relative symlinks are more common here:

```suggestion
    ln -s libFNA3D.so $out/lib/libFNA3D.so.0
    ln -s libFNA3D.so $out/lib/libFNA3D.so.0.${version}
```

File: pkgs/by-name/fn/fna3d/package.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, SDL2
+}:
+stdenv.mkDerivation rec {
+  pname = "fna3d";
+  version = "23.11";
```
Comment (Scrumplex) [inline pkgs/by-name/fn/fna3d/package.nix:9]: Latest version is 24.11

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**